Add configuration file option for passing parameters to dh_python3.

nuclearsandwich opened the pull request:
I've been using this patch in order to pass `--no-guessing-deps` through dh_python3.

Since I've not been using this for Python2 releases I haven't added a sibling configuration value for dh-python2-parameters but it would be straightforward to add.

nuclearsandwich committed:
Add dh_python3_params to override_dh_python3.

len() function isn't required.

In Python3 both of these values convert/coerce to bool as False when
empty so taking the length is not required and is not in style.

A minor change but a reasonable update.

nuclearsandwich committed:
Add configuration option for passing parameters to dh_python3.

nuclearsandwich commented:
I am also interested in helping to maintain stdeb if you're looking for additional maintainers since we're still using it in the Python infrastructure packages for [ROS](https://www.ros.org).

astraw commented:
I sent you an invite to become a maintainer for this stdeb repo. Another maintainer, @dirk-thomas, was also heavily involved in ROS in case you two want to check notes.

I am not really using stdeb much anymore and will leave it to the "real maintainers" (like Dirk and now you) to decide on this.

nuclearsandwich commented:
> I sent you an invite to become a maintainer for this stdeb repo.

Thank you for continuing to support stdeb by responding so quickly to this request!
I am not planning to rush through a new release but if you'd like to get it all taken care of together, I'd like to have access to push future [releases to PyPI](https://pypi.org/project/stdeb/ as well. My username is [nuclearsandwich](https://pypi.org/user/nuclearsandwich/) there as well and I have multi-factor authentication enabled.

> Another maintainer, @dirk-thomas, was also heavily involved in ROS in case you two want to check notes.

Indeed he was. Although his work has taken him elsewhere recently. I'll converse with the rest of the active ROS infrastructure maintainers and suggest a colleague from among them who can help me co-maintain and provide review for this PR in particular.

If there are any other stdeb users who would like to review from the community feedback is very welcome as well!

astraw commented:
@nuclearsandwich PyPI should now also have sent you an invite to be a maintainer of https://pypi.org/project/stdeb/ . Thanks much to you and the ROS team past and present for maintaining this and let me know if I can help in any way.

Evidlo commented:
Any news on the new maintainership?  I'd like to make a PR to be able to specify the signing key, but I'd like to know if it'll get merged.

nuclearsandwich commented:
> Any news on the new maintainership? I'd like to make a PR to be able to specify the signing key, but I'd like to know if it'll get merged.

I have access to review, merge, and make new releases. My time for stdeb will come in small bursts of activity so if you have a PR staged my advice is to open it so that it's there for the next time I can slot some hours for stdeb.

nuclearsandwich committed:
Add a function for running additional test packages.

I didn't want to copy paste this multiple times. I do think there's some
specialized handling for the existing packages so I don't plan to
refactor them to use this function but for adding additional tests this
will save effort.

nuclearsandwich committed:
We can't run python2 package builds without python2.

nuclearsandwich committed:
Add test_data for setuptools and pre-configured packages.

I didn't realize that I added this setuptools package, which explains
why it wasn't part of any tests yet.

I also added a package with an stdeb configuration in order to test
dh-python3-params.

nuclearsandwich committed:
Actually put a run-depend in the config.

cottsay commented on stdeb/util.py:
```diff
@@ -1205,18 +1208,22 @@ def __init__(self,
             'scripts': scripts
         }
 
+        scripts = ''
```
It might be better to keep the type consistent. I'm not sure if mypy is used here, but it would likely complain about this kind of thing.
```suggestion
        scripts = ()
```

nuclearsandwich replied:
I gave this a shot but it caused failures which I didn't have energy to investigate further at the moment.

cottsay reviewed (approved):
One non-blocking comment :+1:

nuclearsandwich committed:
Add dh-python3-params to release notes.
